### DISCUSSION TITLE
Is it possible to switch storage depending on the state between local and session

### DISCUSSION THREAD

**Discussion by qvazi** [pmndrs/zustand#1699] [Q&A]
If the user selected "remember me", then we save the data to the local store, otherwise, to the session
```
import { create } from 'zustand'
import { persist, createJSONStorage } from 'zustand/middleware'

const useMyStore = create(
  persist(
    (set, get) => ({
      isRememberMe: true,
      setStorage: (value) => set(() => ({ isRememberMe: value }))
    }),
    {
      name: 'auth', // unique name
      storage: createJSONStorage((state) => (state.isRememberMe? localStorage : sessionStorage)) 
    }
  )
)
```

**Comment by dai-shi** [comment]
No, but it's an interesting idea!
I hope you or someone would create a third-party library.

**Comment by mapten** [comment]
I am doing it, and it is working. It is explained here:

https://docs.pmnd.rs/zustand/integrations/persisting-store-data#setoptions

You can use this in your component:

```
useEffect(() => {
  useMyStore.persist.setOptions({
    storage: createJSONStorage(() => (isRememberMe ? localStorage : sessionStorage))
  })
}, [isRememberMe]);
```

`isRememberMe` sould belong to another store, like `useConfigurationStore`, with `localStorage` persist option. Or to the component state. But not to the same store. Why? because this value would be overriten to it's default at any point. Or even worse, you can create an infinite loop.

**Reply by vaynevayne** [reply]
Hello, will setOptions make both store have a state?
if i use localStorage, old name is  old, when i  change name by 
`useMyStore.persist.setOptions({
  name: 'new-name',
})`
old storage  name and content is still exists?  

i want set multiple storeName and storeContent  , the i can toggle  it by ` setOptions({
  name: 'new-name',
})`

Is it possible to do so?

**Reply by dbritto-dev** [reply]
> Hello, will setOptions make both store have a state?
> if i use localStorage, old name is  old, when i  change name by 
> `useMyStore.persist.setOptions({
>   name: 'new-name',
> })`
> old storage  name and content is still exists?  
> 
> i want set multiple storeName and storeContent  , the i can toggle  it by ` setOptions({
>   name: 'new-name',
> })`
> 
> Is it possible to do so?

Yeah, it's possible. You need to change the name as well with `persist.setOptions()`

**Reply by vaynevayne** [reply]
thanks
